src/engine.ml

```diff
@@ -1681,10 +1722,7 @@ let incoming ?(is_not_taken = fun _ip -> false) state buf =
                     | Expect_reset, Hard_reset_server_v2 ->
                         let hdr = Packet.header p in
                         Log.warn (fun m ->
-                            m "Hard_reset_client_v2 data: %a" Cstruct.hexdump_pp
-                              data);
-                        Log.warn (fun m ->
-                            m "fixing their_packet_id: %lx" hdr.packet_id);
+                            m "fixing their_packet_id: %08lx" hdr.packet_id);
                         { state.session with their_packet_id = hdr.packet_id }
```

Comment:
It turns out we *don't* fix `their_packet_id` if we send the special packet id ourselves. I'm not sure what to do about this code...